WalletWasabi.Fluent/ViewModels/StatusIcon/HttpGetStringReader.cs

```diff
@@ -0,0 +1,25 @@
+using System.Net.Http;
+using System.Threading.Tasks;
+using WalletWasabi.Fluent.AppServices.Tor;
+using WalletWasabi.WebClients.Wasabi;
+
+namespace WalletWasabi.Fluent.ViewModels.StatusIcon;
+
+public class HttpGetStringReader : IHttpGetStringReader
+{
+	private readonly IWasabiHttpClientFactory _factory;
+
+	public HttpGetStringReader(IWasabiHttpClientFactory wasabiHttpClientFactory)
+	{
+		_factory = wasabiHttpClientFactory;
+	}
+
+	public async Task<string> Read(Uri uri)
+	{
+		using var httpRequestMessage = new HttpRequestMessage(HttpMethod.Get, uri);
+		var httpClient = _factory.NewHttpClientWithDefaultCircuit();
```

Comment:
Well, It's the responsibility of the factory to deal with instances of the client. Following the official docs you will find that `HttpClient`s are constantly created and disposed (per-request, yes). That is not concern in term of performance or resources usage because `HttpMessageHandler` is managed by the factory, and reused (shared).

I imagine that the custom factory we doesn't manage resources like the implementation provided by Microsoft.

Will take a look your code to implement the same solution, then :)

Reply:
> So you are imo incorrect and I have spent non-trivial time trying to come up with

Sorry, let me correct myself. 
What I mean is that the method `IHttpClientFactory.CreateHttpClient` method use thought to be called as many times as the application needs, because the factory will handle the lifetime of the instances to avoid socked exhaustion.